generators/client/templates/react/src/main/webapp/app/modules/administration/metrics/metrics-modal.tsx.ejs

```diff
@@ -1,57 +1,161 @@
 import * as React from 'react';
-import { Table, Modal, ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap';
-
-const MetricsModal = ({ handleClose, showModal, threadDump }) => (
-  <Modal
-    isOpen={showModal} modalTransition={{ timeout: 20 }} backdropTransition={{ timeout: 10 }}
-    toggle={handleClose} className="modal-lg"
-  >
-    <ModalHeader toggle={handleClose}>Thread Dump Info</ModalHeader>
-    <ModalBody>
-      {threadDump && threadDump.threads ? threadDump.threads.map((threadDumpInfo, i) => (
-        <div key={`dump-${i}`}>
-          <h5>{threadDumpInfo.threadState} {threadDumpInfo.threadName} (ID {threadDumpInfo.threadId})</h5>
-          <div className="row" >
-            <div className="col-xs-12">
-              {Object.entries(threadDumpInfo.stackTrace).map(([ stK, stV ]) => (
-                <p key={`detail-${stK}`}>
-                  {stV.className}.{stV.methodName}
-                  ({stV.fileName}:{stV.lineNumber})
-                </p>
-              ))}
-            </div>
-          </div>
-          <div className="row">
-            <div className="col-xs-12">
-              <Table responsive>
-                <thead>
-                  <tr>
-                    <th>Blocked Time</th>
-                    <th>Blocked Count</th>
-                    <th>Waited Time</th>
-                    <th>Waited Count</th>
-                    <th>Lock Name</th>
-                  </tr>
-                </thead>
-                <tbody>
-                  <tr key={threadDumpInfo.lockName}>
-                    <td>{threadDumpInfo.blockedTime}</td>
-                    <td>{threadDumpInfo.blockedCount}</td>
-                    <td>{threadDumpInfo.waitedTime}</td>
-                    <td>{threadDumpInfo.waitedCount}</td>
-                    <td>{threadDumpInfo.lockName}</td>
-                  </tr>
-                </tbody>
-              </Table>
-            </div>
-          </div>
+import { Table, Modal, ModalHeader, ModalBody, ModalFooter, Button, Input, Badge } from 'reactstrap';
+
+import ThreadItem from './thread-item';
+
+export interface IMetricsModalProps {
+  showModal: boolean;
+  handleClose: Function;
+  threadDump: any;
+}
+
+export interface IMetricsModalState {
+  badgeFilter: string;
+  searchFilter: string;
+}
+
+export class MetricsModal extends React.Component<IMetricsModalProps, IMetricsModalState> {
+  state: IMetricsModalState = {
+    badgeFilter: '',
+    searchFilter: ''
+  };
+
+  computeFilteredList = () => {
+    const { badgeFilter, searchFilter } = this.state;
+    let filteredList = this.props.threadDump.threads;
+    if (badgeFilter !== '') {
+      filteredList = filteredList.filter(t => t.threadState === badgeFilter);
+    }
+    if (searchFilter !== '') {
+      filteredList = filteredList.filter(t => t.lockName != null &&
```

Comment:
`t.lockName && t.lockName.toLowerCase().includes(searchFilter.toLowerCase()` would be better